ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceAppliedChanges.mm

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKTransactionalComponentDataSourceAppliedChanges.h"
+#import "CKTransactionalComponentDataSourceAppliedChanges+Internal.h"
+
+#import "ComponentUtilities.h"
+#import "CKInternalHelpers.h"
+#import "CKMacros.h"
+
+@implementation CKTransactionalComponentDataSourceAppliedChanges
+
+- (instancetype)initWithUpdatedIndexPaths:(NSSet *)updatedIndexPaths
+                        removedIndexPaths:(NSSet *)removedIndexPaths
+                          removedSections:(NSIndexSet *)removedSections
+                          movedIndexPaths:(NSDictionary *)movedIndexPaths
+                         insertedSections:(NSIndexSet *)insertedSections
+                       insertedIndexPaths:(NSSet *)insertedIndexPaths
+                                userInfos:(NSArray *)userInfos
```

Comment:
I'm not a big fan of userInfos, could we restrict this to IDs that are returned when a changeset is enqueued and then communicated at the output of the system. This way any metadata management can be done outside of the system itself.